packages/@o3r/design/schematics/generate-css/index.ts

```diff
@@ -0,0 +1,50 @@
+import type { GenerateCssSchematicsSchema } from './schema';
+import type { Rule } from '@angular-devkit/schematics';
+import { globInTree } from '@o3r/schematics';
+import { parseDesignTokenFile, renderDesignTokens } from '@o3r/design';
+import type { DesignTokenRendererOptions, DesignTokenVariableSet, DesignTokenVariableStructure } from '@o3r/design';
+
+/**
+ * Generate CSS from Design Token files
+ * @param options
+ */
+export function generateCss(options: GenerateCssSchematicsSchema): Rule {
+  return async (tree, context) => {
+    const writeFile = (filePath: string, content: string) => tree.exists(filePath) ? tree.overwrite(filePath, content) : tree.create(filePath, content);
+    const readFile = tree.readText;
+    const existsFile = tree.exists;
+    const determineFileToUpdate = options.output ? () => options.output! :
+      (token: DesignTokenVariableStructure) => {
+        if (token.extensions.o3rTargetFile && tree.exists(token.extensions.o3rTargetFile)) {
+          return token.extensions.o3rTargetFile;
+        }
+
+        return options.defaultStyleFile;
+      };
+    const renderDesignTokenOptions: DesignTokenRendererOptions = {
+      readFile,
+      writeFile,
+      existsFile,
+      determineFileToUpdate,
+      logger: context.logger
+    };
+
+    const files = globInTree(tree, Array.isArray(options.designTokenFilePatterns) ? options.designTokenFilePatterns : [options.designTokenFilePatterns]);
+
+    const duplicatedToken: DesignTokenVariableStructure[] = [];
```

Comment:
This code is duplicated in the later schematics - why is that?

Reply:
It is because the purpose of the Builder and the Schematics is not the same and did not want to have a shared function trying to mutualize both things that will evolve in different way.